compiler-core/src/ast.rs

```diff
@@ -406,19 +400,15 @@ pub struct Import<PackageName> {
     pub documentation: Option<SmolStr>,
     pub location: SrcSpan,
     pub module: SmolStr,
-    pub as_name: Option<ImportName>,
+    pub as_name: Option<(AssignName, SrcSpan)>,
     pub unqualified_values: Vec<UnqualifiedImport>,
     pub unqualified_types: Vec<UnqualifiedImport>,
     pub package: PackageName,
 }
 
 impl<T> Import<T> {
-    pub(crate) fn used_name(&self) -> SmolStr {
-        self.as_name
-            .as_ref()
-            .map(|as_name| as_name.name.clone())
-            .or_else(|| self.module.split('/').last().map(|s| s.into()))
-            .expect("Import could not identify variable name.")
+    pub(crate) fn used_name(&self) -> Option<(AssignName, SrcSpan)> {
+        self.as_name.clone()
```

Comment:
This function should still return the used name, as the name says. If other code wants to access the AssignName they can do `import.as_name`